tests/test_benchmarks.py

```
@@ -27,6 +28,18 @@
 
     TORCH_AVAILABLE = True
     CUDA_AVAILABLE = torch.cuda.is_available()
+    # Set CUDA device from environment variable if specified
+    cuda_env = os.environ.get("PYISOLATE_CUDA_DEVICE")
+    if CUDA_AVAILABLE and cuda_env is not None:
+        torch.cuda.set_device(int(cuda_env))
+        print(f"[PyIsolate] Using CUDA device {cuda_env}: {torch.cuda.get_device_name(int(cuda_env))}")
+    elif CUDA_AVAILABLE:
+        print(
+            f"[PyIsolate] Using default CUDA device {torch.cuda.current_device()}: "
+            f"{torch.cuda.get_device_name(torch.cuda.current_device())}"
+        )
+    else:
+        print("[PyIsolate] CUDA not available, using CPU only.")
```

Similar to previous comments, let's just fail in this case instead of silently falling back to CPU-only and succeeding.